### PR TITLE
COMMERCE-1632 bunch of BUGs fixed - urgent for HEADLESS

### PR DESCRIPTION
**Pull request by igorbeslic**
Hi Liferay please review... thank you guys!!

### CODE REVIEW

**Comment by ethanbustad** [comment]
ci:test

**Comment by igorbeslic** [comment]
ci:test

**Comment by igorbeslic** [comment]
@ethanbustad Please can you help me to figure out if any of my changes breaks test?

**Comment by ethanbustad** [comment]
@igorbeslic most of the failures are in upstream, but there are some new failures due to your baseline change -- you'll need to change `com.liferay.commerce.api` and `com.liferay.commerce.product.api` dependencies to project dependencies. I _think_ that's all.

File: commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPAttachmentFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ public void checkCPAttachmentFileEntriesByDisplayDate(
 			serviceContext.setCommand(Constants.UPDATE);
 			serviceContext.setScopeGroupId(cpAttachmentFileEntry.getGroupId());
 
-			cpAttachmentFileEntryLocalService.updateStatus(
```
Comment (marco-leo) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPAttachmentFileEntryLocalServiceImpl.java:206]: @igorbeslic this way to call methods is necessary to support service wrapper extension points: https://portal.liferay.dev/docs/7-0/tutorials/-/knowledge_base/t/customizing-liferay-services-service-wrappers.  
Please revert this commit

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CPDefinitionInventoryLocalServiceImpl.java [hunk 2]
```diff
@@ -39,16 +40,24 @@
 
 	@Override
 	public CPDefinitionInventory addCPDefinitionInventory(
-			long cpDefinitionId, String cpDefinitionInventoryEngine,
-			String lowStockActivity, boolean displayAvailability,
-			boolean displayStockQuantity, int minStockQuantity,
-			boolean backOrders, int minOrderQuantity, int maxOrderQuantity,
-			String allowedOrderQuantities, int multipleOrderQuantity,
-			ServiceContext serviceContext)
+			long groupId, long cpDefinitionId,
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CPDefinitionInventoryLocalServiceImpl.java:43]: @igorbeslic you can obtain the groupId from the CPDefinition

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CPDefinitionInventoryLocalServiceImpl.java (hunk 2 above)
Comment (igorbeslic) [reply commerce-service/src/main/java/com/liferay/commerce/service/impl/CPDefinitionInventoryLocalServiceImpl.java:43]: @marco-leo do you think we can change serviceContext.getUuID with something else? cpDefinition.getUuid() or even generate new uuid? I ask to se if we need serviceContext at all??

Comment (marco-leo) [reply commerce-service/src/main/java/com/liferay/commerce/service/impl/CPDefinitionInventoryLocalServiceImpl.java:43]: @igorbeslic even if I don't like that is the portal standard, and I would like to be complainant to it. So when the UUID is needed, we do have to use it for now

**Comment by marco-leo** [comment]
-

**Pull request closed without merging**